Use clang tidy to update codes for readability.

imikejackson opened the pull request:
clang-tidy was run with the following arguments:
-checks='-*,+readability-*,-readability-identifier-naming,modernize-use-equals-default'

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

mmarineBlueQuartz commented on Source/EbsdLib/HKL/H5CtfImporter.cpp:
```diff
@@ -99,15 +99,16 @@ using namespace H5Support_NAMESPACE;
-  m_FileVersion(Ebsd::H5::FileVersion)
+H5CtfImporter::H5CtfImporter()
+:
+
```
The empty line doesn't seem right.  I'm curious why the formatting turned out this way.

mmarineBlueQuartz commented on Source/Plugins/OrientationAnalysis/Gui/Widgets/OrientationWidget.h:
```diff
@@ -65,7 +65,7 @@ class OrientationWidget : public QWidget
 protected slots:
   virtual void updateData(OrientationUtilityCalculator* calculator);
-  void updateAngleMeasurement(bool isRadians);
+  void updateAngleMeasurement(bool isDegrees);
```
Wait a minute, was this supposed to be degrees or radians?  There's nothing in the pull request saying manual changes were made, and this is a major rename.  Please confirm that this is the correct change.

mmarineBlueQuartz commented on Source/Plugins/Statistics/DistributionAnalysisOps/DistributionAnalysisOps.h:
```diff
@@ -61,7 +61,7 @@ class DistributionAnalysisOps
 
     static void determineMaxAndMinValues(std::vector<float>& data, float& max, float& min);
-    static void determineBinNumbers(float& max, float& min, float& numbins, FloatArrayType::Pointer binnumbers);
+    static void determineBinNumbers(float& max, float& min, float& stepsize, FloatArrayType::Pointer binnumbers);
```
These parameter names imply different things.  Are we sure this change is correct?

mmarineBlueQuartz commented on Source/Plugins/SyntheticBuilding/SyntheticBuildingFilters/InsertPrecipitatePhases.h:
```diff
@@ -323,7 +323,7 @@ class SyntheticBuilding_EXPORT InsertPrecipitatePhases : public AbstractFilter
    * @param featureNum Id for the precipitate to be inserted
    */
-  void insert_precipitate(size_t featureNum);
+  void insert_precipitate(size_t gnum);
```
What is the reason for this change?  featureNum tells me a lot more about the meaning and seems to be correct based on the description.  Does g have a specific meaning for precipitates that I should know about?

mmarineBlueQuartz commented on Source/Plugins/SyntheticBuilding/SyntheticBuildingFilters/InsertPrecipitatePhases.h:
```diff
@@ -332,7 +332,7 @@ class SyntheticBuilding_EXPORT InsertPrecipitatePhases : public AbstractFilter
    * @param zc z centroid coordinate
    */
-  void move_precipitate(int32_t featureNum, float xc, float yc, float zc);
+  void move_precipitate(int32_t gnum, float xc, float yc, float zc);
```
Again, why was the parameter name changed?

mmarineBlueQuartz commented on Source/Plugins/SyntheticBuilding/SyntheticBuildingFilters/InsertPrecipitatePhases.h:
```diff
@@ -370,7 +370,7 @@ class SyntheticBuilding_EXPORT InsertPrecipitatePhases : public AbstractFilter
    * @param double_count Determines whether to double count items in bins
    */
-  void determine_currentRDF(int32_t featureNum, int32_t add, bool double_count);
+  void determine_currentRDF(int32_t gnum, int32_t add, bool double_count);
```
What is gnum?  featureNum is a lot more consistent and seems to be correct based on the description.

mmarineBlueQuartz reviewed (commented):
I agree with most of it, but there were a few places that require specific confirmation before I can approve this.  Mainly, why was each individual parameter I commented on changed in header files?  This seems to be done as an automatic process, so I want to make sure someone can look at these and say they are correct.  Largely, a parameter name being changed from isRadians to isDegrees is a very major change.  If this needs to be changed (and it should at least be brought attention to, since the .h and .cpp disagreed with each other), has this method been used correctly?  Other important points include:

- numbins being replaced with stepsize is another flag.  These do not mean the same thing.  Again, which is correct, and have we been using the method correctly?

- featureNum to gnum: What is a gnum?  I am not familiar with this, so it tells me nothing.  featureNum is easy to read and self-descriptive.  It also matches what the description says is being used.  Was this simply a change based on gnum being shorter to type, or is this another change in meaning?

- OrientationLib changed a lot of (double e0, double e1, double phi2...) into (double phi1, double phi, double phi2...).  I would assume e and phi are referencing different angles or intersections, hence they had been named different things.  Are we absolutely certain the new naming is correct and meaningful?

- OrientationLib also changed euler to synea a few times.  Considering one of the method names was randomizeEulerAngles, how is synea related to the euler angles being randomized?

- A lot of places where QModelIndex& parent was a parameter, parent was replaced with index.  If you look at the methods being overridden, technically parent was the correct name as it was consistent with the virtual method being referenced.  The .cpp, not the .h should have been changed.  This, however, is pretty minor and could be let through without worry.

imikejackson commented:
Those are great observations. I am going to go back and try this whole thing again. the renaming did not happen in the direction that I wanted it to. I was hoping h->cpp and not cpp-h. 

Maybe we can just run that check by itself to at least flag the methods that do not agree with their .cpp counterpart and do some manual updating to ensure they are correct.

I will probably back out this entire PR and start over any maybe not grab so many changes in one shot.

imikejackson committed:
Use clang tidy to update codes for readability.

clang-format was also run on all changed codes

clang-tidy was run with the following arguments:
-checks=-*
,modernize-use-equals-default
,readability-avoid-const-params-in-decls
,readability-braces-around-statements
,readability-container-size-empty
,readability-delete-null-pointer
,readability-else-after-return
,readability-implicit-bool-conversion
,readability-redundant-control-flow
,readability-redundant-declaration
,readability-redundant-function-ptr-dereference
,readability-redundant-member-init
,readability-redundant-smartptr-get
,readability-redundant-string-cstr
,readability-redundant-string-init
,readability-simplify-boolean-expr
,readability-simplify-subscript-expr
,readability-static-accessed-through-instance
,readability-static-definition-in-anonymous-namespace
,readability-string-compare

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

imikejackson commented:
@mmarineBlueQuartz I started a fresh run of clang-tidy and removed some of the checks that were causing issues.

mmarineBlueQuartz reviewed (approved):
This looks good.
